models/repo_list.go

```diff
@@ -217,7 +279,7 @@ func Repositories(opts *SearchRepoOptions) (_ RepositoryList, count int64, err e
 
 	if err = x.
 		Limit(opts.PageSize, (opts.Page-1)*opts.PageSize).
-		OrderBy(opts.OrderBy).
+		OrderBy(opts.OrderBy.String()).
```

Comment:
I think no. `opts.OrderBy` is custom type `SearchOrderBy`, you can't pass just string and it can't be of zero length here, because there is check for that above:
```
+	if len(opts.OrderBy) == 0 {
 -		cond = cond.And(builder.Eq{"is_private": false})		 +		opts.OrderBy = SearchOrderByAlphabetically
  	}		  	}
```

Reply:
But in https://github.com/Morlinest/gitea/blob/0baae9f69e2e54eb4d9fb3329b188f5f9a88bcd9/models/repo_list.go#L275
you it gets set to `"id ASC"` before calling `.String()`

Reply:
You are right, I was looking at other place... Answer: It will not fail :). You can put any string there.